account_cash_invoice/wizard/cash_invoice_in.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2017 Creu Blanca
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class CashInvoiceIn(models.TransientModel):
+    _name = 'cash.invoice.in'
+    _inherit = 'cash.box.out'
+
+    def _default_value(self, default_function):
+        active_model = self.env.context.get('active_model', False)
+        if active_model:
+            active_ids = self.env.context.get('active_ids', False)
+            return default_function(active_model, active_ids)
+        return None
+
+    def _default_company(self):
+        return self._default_value(self.default_company)
+
+    def _default_currency(self):
+        return self._default_value(self.default_currency)
+
+    def _default_journals(self):
+        return self._default_value(self.default_journals)
+
+    def _default_journal(self):
+        journals = self._default_journals()
+        if journals and len(journals.ids) > 0:
+            return self.env['account.journal'].browse(
+                journals.ids[0]
+            ).ensure_one()
+
+    def _default_journal_count(self):
+        return len(self._default_journals().ids)
+
+    invoice_id = fields.Many2one(
+        'account.invoice',
```

Comment:
Non-blocking, but we can have "comodel_name"
Note: I see it is becoming standard in v12.